subprojects/core/src/integTest/groovy/org/gradle/api/tasks/DeferredTaskDefinitionIntegrationTest.groovy

```diff
@@ -758,4 +758,21 @@ class DeferredTaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         then:
         outputContains("got it 15")
     }
+
+    def "configure rule can create additional tasks"() {
+        buildFile << '''
+            def fooTasks = tasks.withType(SomeTask)
+            
+            tasks.createLater('foo', SomeTask) {
+                dependsOn tasks.createLater('bar')
```

Comment:
I think ultimately, we don't want this to work (https://github.com/gradle/gradle-native/issues/709) -- but I couldn't figure out a way to reproduce the CME I was seeing in the other build.  I assume that it's something like this (one task creating another) happening in an indirect way.